lib/config/fileconf_test.go

```diff
@@ -331,6 +332,74 @@ func TestAuthenticationSection(t *testing.T) {
 	}
 }
 
+func TestAuthenticationConfig_HandleSecondFactorOffOnWithoutQoutes(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		desc               string
+		input              string
+		expectError        require.ErrorAssertionFunc
+		expectSecondFactor require.ValueAssertionFunc
+	}{
+		{desc: "handle off with quotes", input: `
+auth_service:
+  enabled: yes
+  authentication:
+    type: local
+    second_factor: "off"
+teleport:
+  nodename: testing
+ssh_service:
+  enabled: yes`,
+			expectError:        require.NoError,
+			expectSecondFactor: requireEqual(constants.SecondFactorOff)},
+		{desc: "handle off without quotes", input: `
+auth_service:
+  enabled: yes
+  authentication:
+    type: local
+    second_factor: off
+teleport:
+  nodename: testing
+ssh_service:
+  enabled: yes`,
+			expectError:        require.NoError,
+			expectSecondFactor: requireEqual(constants.SecondFactorOff)},
+		{desc: "handle on without quotes", input: `
+auth_service:
+  enabled: yes
+  authentication:
+    type: local
+    second_factor: on
+teleport:
+  nodename: testing
+ssh_service:
+  enabled: yes`,
+			expectError:        require.NoError,
+			expectSecondFactor: requireEqual(constants.SecondFactorOn)},
+		{desc: "unsupported numeric type as second_factor", input: `
+auth_service:
+  enabled: yes
+  authentication:
+    type: local
+    second_factor: 4.4
+teleport:
+  nodename: testing
+ssh_service:
+  enabled: yes`,
+			expectError: require.Error,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			cfg, err := ReadConfig(strings.NewReader(tt.input))
+			tt.expectError(t, err)
+			if tt.expectSecondFactor != nil {
```

Comment:
This nil check feels a little weird given the value is being included in each test case, would it be simpler to just have `expectSecondFactor: SecondFactorType` as the field type, and just call `require.Equals(t, tt.expectSecondFactor, cfg.Auth.Authentication.SecondFactor)`

Reply:
it's added because in last test case `unsupported numeric type as second_factor`, there will be error so I am using that nil check to skip value checking in that test case.

Reply:
Ah alright, that makes sense.